calyx/src/passes/well_formed.rs

```diff
@@ -78,6 +80,37 @@ where
     Ok(())
 }
 
+fn same_binding(binding_out: &SmallVec<[(ir::Id, u64); 5]>, binding_in:&SmallVec<[(ir::Id, u64); 5]>) -> bool {
+    if (*binding_out).len() != (*binding_in).len() {
+        return false;
+    }
+
+    //println!("{}",(*binding_out).len());
+
+    for i in 0 .. (*binding_out).len() {
+        let (k_out, v_out) = (*binding_out).get(i).unwrap();
+        let (k_in, v_in) = (*binding_in).get(i).unwrap();
+        if !k_in.eq(k_out) {return false;}
+        if v_in != v_out {return false;}
+    }
+     true
+}
+
+fn same_type(proto_out: &CellType, proto_in: &CellType) -> bool {
+    match proto_out {
```

Comment:
You can simplify this a bit by matching on (proto_out, proto_in) and only handling cases when they are both the same